docs/guides/src/main/server/configuration_overview.adoc

```diff
@@ -0,0 +1,177 @@
+
+In addition to the optimizations performed when you run the `build` command, you might want to avoid using CLI options when running the
+`start` command in favor of using environment variables or configuration options from the `conf/keycloak.conf` file.
+
+.Set any build option
+<@kc.build parameters="--db=postgres"/>
+
+.Set any configuration option to `conf/keycloak.conf`
+```
+db-url-host=keycloak-postgres
+db-username=keycloak
+db-password=change_me
+hostname mykeycloak.acme.com
+```
+
+.Start the server
+<@kc.start/>
+
+By following that approach, the server is going to skip some steps at startup and start faster.
+
+== Using the `auto-build` option when starting the server
+
+The `auto-build` option allows you to perform the two configuration stages using a single command. Under certain circumstances, you might want to sacrifice the server startup time and update the values of build options when starting the server.
+
+For that, you can start the server as follows:
+
+.Using the `auto-build` option
+<@kc.start parameters="--auto-build --db postgres --db-url-host keycloak-postgres --db-username keycloak --db-password change_me --hostname mykeycloak.acme.com"/>
+
+By using this option, the server is going to calculate the build options that have changed and automatically run the `build` command, if necessary, before actually starting the server.
+
+== Configuring the server using configuration files
+
+By default, the server is going to always fetch any configuration option you set from the `conf/keycloak.conf` file. When you are using a fresh distribution,
+this file holds only the recommended settings for running in production, which are initially commented out.
+
+You can also specify a different configuration file by using the `[-cf|--config-file] option as follows:
+
+<@kc.all rootParameters="-cf myconfig.conf"/>
+
+Changes to any *build option* defined in the `keycloak.conf` file that is targeted for the `build` command is ignored
+if the value differs from the value used to previously run the `build` command. In this case, make sure you run the `build` command again so that
+any build option is updated accordingly.
+
+== Using placeholders
+
+You are able to use placeholders to dynamically change the value of configuration options when starting the server.
+
+For that purpose, you use the following format when defining the value of an option:
+
+```
+db.username=${r"${my.db.user}"}
+```
+
+By using a placeholder such as in the example above, the server is going to try to resolve the `my.db.user` variable from:
+
+* A system property using the same key
+* An environment variable by replacing dots with `_`
+
+You should not use placeholders for build options. The reason is that these options
+can not change once they are persisted into the server image. Prefer using placeholders only for the configurations options available to
+the `start` command.
+
+=== Understanding the default profiles
```

Comment:
```suggestion
=== Understanding development mode and production mode
```

Reply:
that would mean there are / may be multiple production modes / development modes as far as i can tell. @andymunro I am not sure aboout this, though. Want to avoid ambiguity, there are exactly 2 modes, development and production. wdyt, as a native speaker? ;)

1) "Understanding development mode and production mode"
2) "Understanding development and production mode"
3) "Understanding development and production modes"
4) "Understanding development- and production mode"
5) "Understanding development- and production modes" 

These are the options floating through my head, but not sure what is exactly right english tbh